tests/unit/gpu_av_ray_query_positive.cpp

```diff
@@ -0,0 +1,400 @@
+    pipeline.descriptor_set_->WriteDescriptorAccelStruct(0, 1, &tlas.GetDstAS()->handle());
+    pipeline.descriptor_set_->UpdateDescriptorSets();
+
+    m_commandBuffer->begin();
+    vk::CmdBindPipeline(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_COMPUTE, pipeline.pipeline_);
+    vk::CmdBindDescriptorSets(m_commandBuffer->handle(), VK_PIPELINE_BIND_POINT_COMPUTE, pipeline.pipeline_layout_.handle(), 0, 1,
+                              &pipeline.descriptor_set_->set_, 0, nullptr);
+    vk::CmdDispatch(m_commandBuffer->handle(), 1, 1, 1);
+    m_commandBuffer->end();
+
+    m_commandBuffer->QueueCommandBuffer();
+    m_device->wait();
+}
+
+TEST_F(PositiveGpuAV, RayQueryComputeDynamicTminTmax) {
+    TEST_DESCRIPTION("Ray query in a compute shader, with dynamically set t_min and t_max");
+
+    SetTargetApiVersion(VK_API_VERSION_1_2);
+
+    AddRequiredFeature(vkt::Feature::rayQuery);
+    AddRequiredFeature(vkt::Feature::accelerationStructure);
+    AddRequiredFeature(vkt::Feature::bufferDeviceAddress);
+
+    AddRequiredExtensions(VK_KHR_RAY_QUERY_EXTENSION_NAME);
+
+    RETURN_IF_SKIP(InitGpuAvFramework());
+    RETURN_IF_SKIP(InitState());
+
+    char const *shader_source = R"glsl(
+        #version 460
+        #extension GL_EXT_ray_query : require
+        
+        layout(set = 0, binding = 0) uniform accelerationStructureEXT tlas;
+        layout(set = 0, binding = 1) uniform Uniforms {
+          float t_min;
+          float t_max;
+        } trace_rays_params;
+
+        void main() {
+            rayQueryEXT query;
+            rayQueryInitializeEXT(query, tlas, gl_RayFlagsTerminateOnFirstHitEXT, 0xff, vec3(0), 
+              trace_rays_params.t_min, vec3(0,0,1), trace_rays_params.t_max);
+            rayQueryProceedEXT(query);
+            // uint intersection_type = rayQueryGetIntersectionTypeEXT(query, true);
+        }
+    )glsl";
+
+    CreateComputePipelineHelper pipeline(*this);
+    pipeline.cs_ = std::make_unique<VkShaderObj>(this, shader_source, VK_SHADER_STAGE_COMPUTE_BIT, SPV_ENV_VULKAN_1_2);
+    pipeline.dsl_bindings_ = {{0, VK_DESCRIPTOR_TYPE_ACCELERATION_STRUCTURE_KHR, 1, VK_SHADER_STAGE_COMPUTE_BIT, nullptr},
+                              {1, VK_DESCRIPTOR_TYPE_UNIFORM_BUFFER, 1, VK_SHADER_STAGE_COMPUTE_BIT, nullptr}};
+    pipeline.InitState();
+    pipeline.CreateComputePipeline();
+
+    // Add TLAS binding
+    vkt::as::BuildGeometryInfoKHR tlas = vkt::as::blueprint::BuildOnDeviceTopLevel(*m_device, *m_commandBuffer);
+    pipeline.descriptor_set_->WriteDescriptorAccelStruct(0, 1, &tlas.GetDstAS()->handle());
+
+    // Add uniform buffer binding
+    vkt::Buffer uniform_buffer(*m_device, 4096, VK_BUFFER_USAGE_UNIFORM_BUFFER_BIT,
+                               VK_MEMORY_PROPERTY_HOST_VISIBLE_BIT | VK_MEMORY_PROPERTY_HOST_COHERENT_BIT);
+    pipeline.descriptor_set_->WriteDescriptorBufferInfo(1, uniform_buffer, 0, uniform_buffer.create_info().size);
```

Comment:
trick for all these  `WriteDescriptorBufferInfo`
```suggestion
    pipeline.descriptor_set_->WriteDescriptorBufferInfo(1, uniform_buffer, 0, VK_WHOLE_SIZE);
```